### PR TITLE
Refactored editor state fields and methods to EditorState class

### LINKED ISSUE #125: Refactor Editor state to a new class
We currently keep a bunch of flags/fields and methods, that indicate the current state of the Editor, directly in the EditorTopComponent. To avoid this growing too large and also for the better readability, we should refactor this to a new class, called "EditorState" and keep those fields and methods there. 
Those fields include:
```
    private boolean dirty = false;
    private boolean modified = false;
    private boolean previewDisplayed = true;
    private File currentFile;
    private DbxState dbxState;
```
and all should be moved to a new class. The EditorTopComponent should expose a single method, that returns an instance of EditorState class.

### PR DESCRIPTION
**Pull request by andyzen619**
Fixes #125 and updates #138

### CODE REVIEW

**Comment by andyzen619** [comment]
Hey, I accidentally deleted my fork so I had to create another pull request. Hopefully this doesn't cause complications.

**Comment by sebbrudzinski** [comment]
Travis reports that there's a problem with code style. Please check and fix so the build passes.

**Comment by andyzen619** [comment]
Hey @sebbrudzinski , I fixed the check style issue. Everything should be okay now.

File: OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java [hunk 1]
```diff
@@ -148,15 +143,15 @@ public void keyTyped(java.awt.event.KeyEvent evt) {
     }// </editor-fold>                        
 
     private void rSyntaxTextAreaKeyReleased(java.awt.event.KeyEvent evt) {
-        dirty = true;
-        setModified(true);
+        getEditorState().setDirty(true);
```
Comment (sebbrudzinski) [inline OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java]: No need to use getter in this class. Just change it to editorState.setDirty(true). This applies to all "getEditorState()" in THIS class.

**Comment by andyzen619** [comment]
Hey @sebbrudzinski , I have made the changes that you have suggested.

**Pull request merged**